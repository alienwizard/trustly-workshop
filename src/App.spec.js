import React from 'react'
const { render } = require('@testing-library/react')
const { default: App, CakeContext } = require('./App')

test('should render cakes', async () => {
  const { findAllByText } = render(<App />, {
    wrapper: ({ children }) => (
      <CakeContext.Provider>{children}</CakeContext.Provider>
    ),
  })
  const cakeApp = await findAllByText('Test cake')
  expect(cakeApp[0]).toBeInTheDocument()
})

test('should be able to add cake to favorite display', async () => {
  const { findAllByText, getAllByTestId } = render(<App />, {
    wrapper: ({ children }) => (
      <CakeContext.Provider>{children}</CakeContext.Provider>
    ),
  })
  const [favoriteButton] = await findAllByText('Favorite')
  expect(favoriteButton).toBeInTheDocument()
  favoriteButton.click()
  const favorites = getAllByTestId('favorite')

  expect(favorites).toMatchSnapshot()
})

test('should be able to remove cake to favorite display', async () => {
  const { findAllByText, getAllByTestId } = render(<App />, {
    wrapper: ({ children }) => (
      <CakeContext.Provider>{children}</CakeContext.Provider>
    ),
  })
  const [favoriteButton] = await findAllByText('Favorite')
  expect(favoriteButton).toBeInTheDocument()
  favoriteButton.click()

  const [favorite] = getAllByTestId('favorite')
  expect(favorite).toBeInTheDocument()

  favoriteButton.click()

  expect(favorite).not.toBeInTheDocument()
})
