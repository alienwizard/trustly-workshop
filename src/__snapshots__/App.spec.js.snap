// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to add cake to favorite display 1`] = `
Array [
  <div
    data-testid="favorite"
  >
    Test cake
  </div>,
]
`;

exports[`should be able to add cake to favorite display 2`] = `
Array [
  <div
    data-testid="favorite"
  >
    Test cake
  </div>,
]
`;
